In [1]:
pip install fredapi

     |████████████████████████████████| 9.9 MB 3.9 MB/s eta 0:00:01
     |████████████████████████████████| 14.8 MB 36.6 MB/s eta 0:00:01
     |████████████████████████████████| 510 kB 39.1 MB/s eta 0:00:01
Note: you may need to restart the kernel to use updated packages.


In [2]:
import pandas as pd

In [3]:
from fredapi import Fred

In [4]:
import json

In [5]:
import requests

In [6]:
import datetime

# 1.1. Extracting Information from my own transaction

In [7]:
answer2 = requests.get("https://api.whatsonchain.com/v1/bsv/main/block/height/669069")
bitcointransfer_block = json.loads(answer2.text)

In [8]:
bitcointransfer_block

{'hash': '00000000000000000b9764acf9b16d0a9010dec087d95d2ae39b260d95b7cc5b',
 'confirmations': 938,
 'size': 1590970,
 'height': 669069,
 'version': 939515904,
 'versionHex': '37ffe000',
 'merkleroot': 'c48b31ec0167278643e4b7ebbaf2fddde6575999a0d9c656340746edc7823737',
 'txcount': 5354,
 'tx': ['478a300c85e3274be516482a0f1d1229f27a97c74c2c8032aedc9298a8a1494d',
  '803b6bdfaf0df724ce350920b63f99ed6aaa1a9f553aec7fb2ee48f2269e9606',
  '9dff11264545f427f3c46ef33e8db740ba3e1f522ecd39eb178bb64efb6edd21',
  'adf6a735617b431773dab480220a75d3d929253b30a0f6ebc0d2c29a5bc55441',
  '9002d1cf0a402c4b9349498da8a4ca106ae38c3ded83acc1d8c9cd9634217386',
  'b3913983f89d0d4033bf04ec75c2115f5f7b497951f061901d3df2c6a8876cd1',
  '8c1170ba842e2537041ad50a88e9d3e2c5af03184e2fc5e8ca2a000dfd97e671',
  'aa26e94a8eab44e6e94240629031d10ff337898321f91a8f9940255ff8ede086',
  '64ac77fdd64697ba5615be4eff0d5b3b923e2efb23d2278c5c916459e28948d3',
  '981ba3d1b4315f96d8de8128e3d9673bb3aa3145f6135f0a2f7b473385695994',
  '402

In [9]:
print(bitcointransfer_block['miner'])

taal.com


In [ ]:
# Miners records every transactions to the blockchain. Here the miner is called taal.com

In [10]:
print(bitcointransfer_block['txcount'])

5354


In [ ]:
# 5354 transactions in total since the transaction begins

In [11]:
print(bitcointransfer_block['totalFees'])

0.009363360000000043


In [12]:
print(bitcointransfer_block['confirmations'])

938


In [ ]:
# "confirmation" in Bitcoin refers the number blocks has been created since the transaction has taken place

In [13]:
timestamp = datetime.datetime.fromtimestamp(bitcointransfer_block['time'])

In [14]:
print(timestamp.strftime('%Y-%m-%d %H:%M:%S'))

2021-01-08 11:08:24


In [ ]:
# timestamp shows the time of the bitcoin transaction. Here itis on 8th January 2021 at 11:08

# 1.2 Extract Information from Famous Blocks

The First ever transaction from Satoshi to Hal Finney in 2010


In [15]:
answer = requests.get("https://api.whatsonchain.com/v1/bsv/main/block/height/170")
bitcoin_block = json.loads(answer.text)

In [16]:
bitcoin_block

{'hash': '00000000d1145790a8694403d4063f323d499e655c83426834d4ce2f8dd4a2ee',
 'confirmations': 669837,
 'size': 490,
 'height': 170,
 'version': 1,
 'versionHex': '00000001',
 'merkleroot': '7dac2c5666815c17a3b36427de37bb9d2e2c5ccec3f8633eb91a4205cb4c10ff',
 'txcount': 2,
 'tx': ['b1fea52486ce0c62bb442b530a3f0132b826c74e473d1f2c220bfa78111c5082',
  'f4184fc596403b9d638783cf57adfe4c75c605f6356fbc91338530e9831e9e16'],
 'time': 1231731025,
 'mediantime': 1231716245,
 'nonce': 1889418792,
 'bits': '1d00ffff',
 'difficulty': 1,
 'chainwork': '000000000000000000000000000000000000000000000000000000ab00ab00ab',
 'previousblockhash': '000000002a22cfee1f2c846adbd12b3e183d4f97683f85dad08a79780a84bd55',
 'nextblockhash': '00000000c9ec538cab7f38ef9c67a95742f56ab07b0a37c5be6b02808dbfb4e0',
 'coinbaseTx': {'hex': '',
  'txid': 'b1fea52486ce0c62bb442b530a3f0132b826c74e473d1f2c220bfa78111c5082',
  'hash': 'b1fea52486ce0c62bb442b530a3f0132b826c74e473d1f2c220bfa78111c5082',
  'version': 1,
  'size': 134,

In [17]:
print(bitcoin_block['miner'])

��


In [18]:
print(bitcoin_block['txcount'])

2


In [19]:
print(bitcoin_block['totalFees'])

0


In [20]:
print(bitcoin_block['confirmations'])

669837


In [21]:
timestamp1 = datetime.datetime.fromtimestamp(bitcoin_block['time'])

In [22]:
print(timestamp1.strftime('%Y-%m-%d %H:%M:%S'))

2009-01-12 03:30:25


The Pizza purchase for 10,000BTC in 2010

In [23]:
answer1 = requests.get("https://api.whatsonchain.com/v1/bsv/main/block/height/57043")
pizza_block = json.loads(answer1.text)

In [24]:
pizza_block 

{'hash': '00000000152340ca42227603908689183edc47355204e7aca59383b0aaac1fd8',
 'confirmations': 612964,
 'size': 23835,
 'height': 57043,
 'version': 1,
 'versionHex': '00000001',
 'merkleroot': '5c1d2211f598cd6498f42b269fe3ce4a6fdb40eaa638f86a0579c4e63a721b5a',
 'txcount': 2,
 'tx': ['bd9075d78e65a98fb054cb33cf0ecf14e3e7f8b3150231df8680919a79ac8fe5',
  'a1075db55d416d3ca199f55b6084e2115b9345e16c5cf302fc80e9d5fbf5d48d'],
 'time': 1274552191,
 'mediantime': 1274548928,
 'nonce': 188133155,
 'bits': '1c159c24',
 'difficulty': 11.84622814970471,
 'chainwork': '00000000000000000000000000000000000000000000000000026c60b6429108',
 'previousblockhash': '0000000013e7e85518dac94d012d73253d3fdac5c30c4143b177f3086f129580',
 'nextblockhash': '0000000013ab9f8ed78b254a429d3d5ad52905362e01bf6c682940337721eb51',
 'coinbaseTx': {'hex': '',
  'txid': 'bd9075d78e65a98fb054cb33cf0ecf14e3e7f8b3150231df8680919a79ac8fe5',
  'hash': 'bd9075d78e65a98fb054cb33cf0ecf14e3e7f8b3150231df8680919a79ac8fe5',
  'version'

In [25]:
print(pizza_block['miner'])

$�v


In [26]:
print(pizza_block['txcount'])

2


In [27]:
print(pizza_block['totalFees'])

0.990000000000002


In [28]:
print(pizza_block['confirmations'])

612964


In [29]:
timestamp2 = datetime.datetime.fromtimestamp(pizza_block['time'])

In [30]:
print(timestamp2.strftime('%Y-%m-%d %H:%M:%S'))

2010-05-22 18:16:31


# 2. Time Series Investigation of Bitcoin Price

In [31]:
fred=Fred(api_key='e3c61d1f1be469c9885504585a56a87c')

In [32]:
bitcoinprice = fred.get_series('CBBTCUSD')

In [33]:
bitcoinprice

2014-12-01      370.00
2014-12-02      378.00
2014-12-03      378.00
2014-12-04      377.10
2014-12-05         NaN
                ...   
2021-01-09    40346.30
2021-01-10    38242.02
2021-01-11    35474.31
2021-01-12    33968.43
2021-01-13    37520.66
Length: 2236, dtype: float64

In [35]:
goldprice = fred.get_series('ID7108')
market = fred.get_series('SP500')
riskfreerate = fred.get_series('TB3MS')

In [36]:
dataframe = pd.concat([bitcoinprice, goldprice, market, riskfreerate], axis=1)

In [37]:
dataframe.columns = ['bitcoinprice', 'goldprice', 'market', 'riskfreerate']

In [39]:
dataframe

bitcoinprice  goldprice   market  riskfreerate
1934-01-01           NaN        NaN      NaN          0.72
1934-02-01           NaN        NaN      NaN          0.62
1934-03-01           NaN        NaN      NaN          0.24
1934-04-01           NaN        NaN      NaN          0.15
1934-05-01           NaN        NaN      NaN          0.16
...                  ...        ...      ...           ...
2021-01-09      40346.30        NaN      NaN           NaN
2021-01-10      38242.02        NaN      NaN           NaN
2021-01-11      35474.31        NaN  3799.61           NaN
2021-01-12      33968.43        NaN  3801.19           NaN
2021-01-13      37520.66        NaN  3809.84           NaN

[4186 rows x 4 columns]

In [40]:
dataframe.to_csv('btc_analysis.csv')

In [41]:
dataframe = dataframe.fillna(method='ffill')

In [57]:
dataframe = dataframe.loc['2016-01-01':'2021-01-14']

In [58]:
dataframe

bitcoinprice  goldprice   market  riskfreerate       Rbt  \
2016-02-01        371.33      412.4  1939.38          0.31 -0.159771   
2016-03-01        433.08      451.2  1978.35          0.29  0.153831   
2016-04-01        417.68      445.4  2072.78          0.23 -0.036207   
2016-05-01        456.98      465.7  2065.30          0.27  0.089924   
2016-06-01        534.84      441.3  2099.33          0.27  0.157328   
2016-07-01        678.19      490.7  2102.95          0.30  0.237460   
2016-08-01        605.20      490.1  2170.84          0.30 -0.113869   
2016-09-01        573.02      484.6  2170.86          0.29 -0.054638   
2016-10-01        615.65      461.2  2168.27          0.33  0.071758   
2016-11-01        730.70      467.6  2111.72          0.45  0.171324   
2016-12-01        754.00      422.7  2191.08          0.51  0.031389   
2017-01-01        992.95      421.9  2238.83          0.51  0.275288   
2017-02-01        992.75      437.2  2279.55          0.52 -0.000201   
2017-03-01       1230.00      449.6  2395.96          0.74  0.214291   
2017-04-01       1092.00      453.3  2362.72          0.80 -0.119003   
2017-05-01       1436.50      450.4  2388.33          0.89  0.274199   
2017-06-01       2419.99      463.0  2430.06          0.98  0.521554   
2017-07-01       2423.63      441.9  2423.41          1.07  0.001503   
2017-08-01       2732.59      457.9  2476.35          1.01  0.119983   
2017-09-01       4947.99      484.7  2476.55          1.03  0.593732   
2017-10-01       4394.81      459.6  2519.36          1.07 -0.118557   
2017-11-01       6783.69      460.4  2579.36          1.23  0.434097   
2017-12-01      10869.84      456.4  2642.22          1.32  0.471471   
2018-01-01      13480.01      475.9  2673.61          1.41  0.215216   
2018-02-01       9014.23      478.7  2821.98          1.57 -0.402403   
2018-03-01      10895.92      479.1  2677.67          1.70  0.189584   
2018-04-01       6816.01      482.0  2640.87          1.76 -0.469114   
2018-05-01       9072.29      473.7  2654.80          1.86  0.285950   
2018-06-01       7514.32      469.0  2734.62          1.90 -0.188414   
2018-07-01       6349.50      453.1  2718.37          1.96 -0.168434   
2018-08-01       7603.99      440.8  2813.36          2.03  0.180297   
2018-09-01       7191.08      433.5  2901.52          2.13 -0.055832   
2018-10-01       6572.83      434.5  2924.59          2.25 -0.089897   
2018-11-01       6344.00      444.0  2740.37          2.33 -0.035435   
2018-12-01       4142.01      448.7  2760.17          2.37 -0.426328   
2019-01-01       3826.10      464.4  2506.85          2.37 -0.079335   
2019-02-01       3437.50      475.9  2706.53          2.39 -0.107102   
2019-03-01       3806.17      467.3  2803.69          2.40  0.101879   
2019-04-01       4137.00      467.4  2867.19          2.38  0.083347   
2019-05-01       5321.15      463.7  2923.73          2.35  0.251719   
2019-06-01       8558.01      481.8  2752.06          2.17  0.475178   
2019-07-01      10570.62      506.1  2964.33          2.10  0.211211   
2019-08-01      10417.24      537.7  2953.56          1.95 -0.014616   
2019-09-01       9769.64      554.7  2926.46          1.89 -0.064183   
2019-10-01       8323.43      541.5  2940.25          1.65 -0.160205   
2019-11-01       9268.00      539.2  3066.91          1.54  0.107493   
2019-12-01       7413.84      533.9  3140.98          1.54 -0.223219   
2020-01-01       7174.32      555.0  3230.78          1.52 -0.032841   
2020-02-01       9375.06      568.1  3225.52          1.52  0.267545   
2020-03-01       8490.30      595.4  2954.22          0.29 -0.099129   
2020-04-01       6719.39      583.4  2470.50          0.14 -0.233927   
2020-05-01       8813.90      617.3  2830.71          0.13  0.271333   
2020-06-01      10212.38      620.8  3055.73          0.16  0.147271   
2020-07-01       9239.09      642.6  3115.86          0.13 -0.100157   
2020-08-01      11805.82      730.8  3271.12        

In [59]:
dataframe = dataframe[dataframe.index.day==1]

In [60]:
dataframe

bitcoinprice  goldprice   market  riskfreerate       Rbt  \
2016-02-01        371.33      412.4  1939.38          0.31 -0.159771   
2016-03-01        433.08      451.2  1978.35          0.29  0.153831   
2016-04-01        417.68      445.4  2072.78          0.23 -0.036207   
2016-05-01        456.98      465.7  2065.30          0.27  0.089924   
2016-06-01        534.84      441.3  2099.33          0.27  0.157328   
2016-07-01        678.19      490.7  2102.95          0.30  0.237460   
2016-08-01        605.20      490.1  2170.84          0.30 -0.113869   
2016-09-01        573.02      484.6  2170.86          0.29 -0.054638   
2016-10-01        615.65      461.2  2168.27          0.33  0.071758   
2016-11-01        730.70      467.6  2111.72          0.45  0.171324   
2016-12-01        754.00      422.7  2191.08          0.51  0.031389   
2017-01-01        992.95      421.9  2238.83          0.51  0.275288   
2017-02-01        992.75      437.2  2279.55          0.52 -0.000201   
2017-03-01       1230.00      449.6  2395.96          0.74  0.214291   
2017-04-01       1092.00      453.3  2362.72          0.80 -0.119003   
2017-05-01       1436.50      450.4  2388.33          0.89  0.274199   
2017-06-01       2419.99      463.0  2430.06          0.98  0.521554   
2017-07-01       2423.63      441.9  2423.41          1.07  0.001503   
2017-08-01       2732.59      457.9  2476.35          1.01  0.119983   
2017-09-01       4947.99      484.7  2476.55          1.03  0.593732   
2017-10-01       4394.81      459.6  2519.36          1.07 -0.118557   
2017-11-01       6783.69      460.4  2579.36          1.23  0.434097   
2017-12-01      10869.84      456.4  2642.22          1.32  0.471471   
2018-01-01      13480.01      475.9  2673.61          1.41  0.215216   
2018-02-01       9014.23      478.7  2821.98          1.57 -0.402403   
2018-03-01      10895.92      479.1  2677.67          1.70  0.189584   
2018-04-01       6816.01      482.0  2640.87          1.76 -0.469114   
2018-05-01       9072.29      473.7  2654.80          1.86  0.285950   
2018-06-01       7514.32      469.0  2734.62          1.90 -0.188414   
2018-07-01       6349.50      453.1  2718.37          1.96 -0.168434   
2018-08-01       7603.99      440.8  2813.36          2.03  0.180297   
2018-09-01       7191.08      433.5  2901.52          2.13 -0.055832   
2018-10-01       6572.83      434.5  2924.59          2.25 -0.089897   
2018-11-01       6344.00      444.0  2740.37          2.33 -0.035435   
2018-12-01       4142.01      448.7  2760.17          2.37 -0.426328   
2019-01-01       3826.10      464.4  2506.85          2.37 -0.079335   
2019-02-01       3437.50      475.9  2706.53          2.39 -0.107102   
2019-03-01       3806.17      467.3  2803.69          2.40  0.101879   
2019-04-01       4137.00      467.4  2867.19          2.38  0.083347   
2019-05-01       5321.15      463.7  2923.73          2.35  0.251719   
2019-06-01       8558.01      481.8  2752.06          2.17  0.475178   
2019-07-01      10570.62      506.1  2964.33          2.10  0.211211   
2019-08-01      10417.24      537.7  2953.56          1.95 -0.014616   
2019-09-01       9769.64      554.7  2926.46          1.89 -0.064183   
2019-10-01       8323.43      541.5  2940.25          1.65 -0.160205   
2019-11-01       9268.00      539.2  3066.91          1.54  0.107493   
2019-12-01       7413.84      533.9  3140.98          1.54 -0.223219   
2020-01-01       7174.32      555.0  3230.78          1.52 -0.032841   
2020-02-01       9375.06      568.1  3225.52          1.52  0.267545   
2020-03-01       8490.30      595.4  2954.22          0.29 -0.099129   
2020-04-01       6719.39      583.4  2470.50          0.14 -0.233927   
2020-05-01       8813.90      617.3  2830.71          0.13  0.271333   
2020-06-01      10212.38      620.8  3055.73          0.16  0.147271   
2020-07-01       9239.09      642.6  3115.86          0.13 -0.100157   
2020-08-01      11805.82      730.8  3271.12        

In [61]:
import numpy as np

In [62]:
dataframe['Rbt'] = np.log(dataframe['bitcoinprice']/dataframe['bitcoinprice'].shift(1))

In [63]:
dataframe['Rgt'] = np.log(dataframe['goldprice']/dataframe['goldprice'].shift(1))
dataframe['Rmt'] = np.log(dataframe['market']/dataframe['market'].shift(1))
dataframe['Rft'] = dataframe['riskfreerate']/4

In [64]:
dataframe

bitcoinprice  goldprice   market  riskfreerate       Rbt  \
2016-02-01        371.33      412.4  1939.38          0.31       NaN   
2016-03-01        433.08      451.2  1978.35          0.29  0.153831   
2016-04-01        417.68      445.4  2072.78          0.23 -0.036207   
2016-05-01        456.98      465.7  2065.30          0.27  0.089924   
2016-06-01        534.84      441.3  2099.33          0.27  0.157328   
2016-07-01        678.19      490.7  2102.95          0.30  0.237460   
2016-08-01        605.20      490.1  2170.84          0.30 -0.113869   
2016-09-01        573.02      484.6  2170.86          0.29 -0.054638   
2016-10-01        615.65      461.2  2168.27          0.33  0.071758   
2016-11-01        730.70      467.6  2111.72          0.45  0.171324   
2016-12-01        754.00      422.7  2191.08          0.51  0.031389   
2017-01-01        992.95      421.9  2238.83          0.51  0.275288   
2017-02-01        992.75      437.2  2279.55          0.52 -0.000201   
2017-03-01       1230.00      449.6  2395.96          0.74  0.214291   
2017-04-01       1092.00      453.3  2362.72          0.80 -0.119003   
2017-05-01       1436.50      450.4  2388.33          0.89  0.274199   
2017-06-01       2419.99      463.0  2430.06          0.98  0.521554   
2017-07-01       2423.63      441.9  2423.41          1.07  0.001503   
2017-08-01       2732.59      457.9  2476.35          1.01  0.119983   
2017-09-01       4947.99      484.7  2476.55          1.03  0.593732   
2017-10-01       4394.81      459.6  2519.36          1.07 -0.118557   
2017-11-01       6783.69      460.4  2579.36          1.23  0.434097   
2017-12-01      10869.84      456.4  2642.22          1.32  0.471471   
2018-01-01      13480.01      475.9  2673.61          1.41  0.215216   
2018-02-01       9014.23      478.7  2821.98          1.57 -0.402403   
2018-03-01      10895.92      479.1  2677.67          1.70  0.189584   
2018-04-01       6816.01      482.0  2640.87          1.76 -0.469114   
2018-05-01       9072.29      473.7  2654.80          1.86  0.285950   
2018-06-01       7514.32      469.0  2734.62          1.90 -0.188414   
2018-07-01       6349.50      453.1  2718.37          1.96 -0.168434   
2018-08-01       7603.99      440.8  2813.36          2.03  0.180297   
2018-09-01       7191.08      433.5  2901.52          2.13 -0.055832   
2018-10-01       6572.83      434.5  2924.59          2.25 -0.089897   
2018-11-01       6344.00      444.0  2740.37          2.33 -0.035435   
2018-12-01       4142.01      448.7  2760.17          2.37 -0.426328   
2019-01-01       3826.10      464.4  2506.85          2.37 -0.079335   
2019-02-01       3437.50      475.9  2706.53          2.39 -0.107102   
2019-03-01       3806.17      467.3  2803.69          2.40  0.101879   
2019-04-01       4137.00      467.4  2867.19          2.38  0.083347   
2019-05-01       5321.15      463.7  2923.73          2.35  0.251719   
2019-06-01       8558.01      481.8  2752.06          2.17  0.475178   
2019-07-01      10570.62      506.1  2964.33          2.10  0.211211   
2019-08-01      10417.24      537.7  2953.56          1.95 -0.014616   
2019-09-01       9769.64      554.7  2926.46          1.89 -0.064183   
2019-10-01       8323.43      541.5  2940.25          1.65 -0.160205   
2019-11-01       9268.00      539.2  3066.91          1.54  0.107493   
2019-12-01       7413.84      533.9  3140.98          1.54 -0.223219   
2020-01-01       7174.32      555.0  3230.78          1.52 -0.032841   
2020-02-01       9375.06      568.1  3225.52          1.52  0.267545   
2020-03-01       8490.30      595.4  2954.22          0.29 -0.099129   
2020-04-01       6719.39      583.4  2470.50          0.14 -0.233927   
2020-05-01       8813.90      617.3  2830.71          0.13  0.271333   
2020-06-01      10212.38      620.8  3055.73          0.16  0.147271   
2020-07-01       9239.09      642.6  3115.86          0.13 -0.100157   
2020-08-01      11805.82      730.8  3271.12        

In [65]:
dataframe = dataframe.drop(dataframe.index[0])

In [66]:
Bitcoinreturns = dataframe["Rbt"]    
Goldreturns = dataframe["Rgt"]
Marketreturns = dataframe["Rmt"]
Riskfreereturns = dataframe["Rft"]

In [67]:
Bit_Gold_correl = Bitcoinreturns.corr(Goldreturns)

In [68]:
Bit_Gold_correl

0.2076859353930392

The correlation between bitcoinreturn and goldreturn is 0.207, which means they are not correlated strongly to each other.

In [69]:
Bit_Market_correl = Bitcoinreturns.corr(Marketreturns)

In [70]:
Bit_Market_correl

0.19089529681473805

Bitcoinreturns and Marketreturns are correlated with each other with a value of 0.1908. That shows the relationship between them is not strong.

In [71]:
!pip3 install scikit-learn

     |████████████████████████████████| 22.3 MB 4.4 MB/s eta 0:00:01
     |████████████████████████████████| 27.4 MB 39.5 MB/s eta 0:00:01
     |████████████████████████████████| 302 kB 44.9 MB/s eta 0:00:01


In [72]:
from sklearn.linear_model import LinearRegression

In [73]:
w = Bitcoinreturns-Riskfreereturns
x = Goldreturns-Riskfreereturns
y = Marketreturns-Riskfreereturns

In [74]:
w.shape

(59,)

In [75]:
w = w.values.reshape(-1, 1)

In [76]:
w.shape

(59, 1)

In [77]:
w

array([[ 0.08133131],
       [-0.09370688],
       [ 0.02242404],
       [ 0.08982801],
       [ 0.16245985],
       [-0.1888685 ],
       [-0.12713836],
       [-0.010742  ],
       [ 0.05882436],
       [-0.09611061],
       [ 0.14778794],
       [-0.13020144],
       [ 0.02929058],
       [-0.31900329],
       [ 0.05169872],
       [ 0.27655381],
       [-0.26599699],
       [-0.13251654],
       [ 0.33623156],
       [-0.38605713],
       [ 0.1265969 ],
       [ 0.14147078],
       [-0.13728413],
       [-0.79490341],
       [-0.23541603],
       [-0.90911415],
       [-0.17904954],
       [-0.66341418],
       [-0.65843446],
       [-0.32720296],
       [-0.58833174],
       [-0.65239688],
       [-0.617935  ],
       [-1.01882831],
       [-0.67183517],
       [-0.70460154],
       [-0.49812103],
       [-0.51165255],
       [-0.33578144],
       [-0.06732176],
       [-0.31378923],
       [-0.50211633],
       [-0.53668251],
       [-0.57270519],
       [-0.27750682],
       [-0

In [78]:
x = x.values.reshape(-1, 1)

In [79]:
y = y.values.reshape(-1, 1)

In [80]:
lr = LinearRegression()

In [81]:
lr.fit(w, y)
print(lr.intercept_, lr.coef_)

[-0.17190718] [[0.4791191]]


The intercept for Bitcoinreturns is -0.1719 and the slope coefficient is 0.4791

In [82]:
lr.fit(x, y)
print(lr.intercept_, lr.coef_)

[-0.00308916] [[0.97690782]]


The intercept  for goldreturn is -0.0030 and the slope coefficent is 0.9769